mavros_extras/src/plugins/vision_pose_estimate.cpp

```diff
@@ -15,57 +15,82 @@
+#include "rcpputils/asserts.hpp"
+#include "mavros/mavros_uas.hpp"
+#include "mavros/utils.hpp"
+#include "mavros/plugin.hpp"
+#include "mavros/plugin_filter.hpp"
+#include "mavros/setpoint_mixin.hpp"
+
+#include <tf2_eigen/tf2_eigen.h>
+
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "geometry_msgs/msg/pose_with_covariance_stamped.hpp" //
+#include "geometry_msgs/msg/vector3_stamped.hpp"
+#include "mavros_msgs/msg/landing_target.hpp"
+
 
-#include <geometry_msgs/PoseStamped.h>
-#include <geometry_msgs/PoseWithCovarianceStamped.h>
 
 namespace mavros
 {
 namespace extra_plugins
 {
+using namespace std::placeholders;
+
 /**
  * @brief Vision pose estimate plugin
+ * @plugin vision_pose
  *
  * Send pose estimation from various vision estimators
  * to FCU position and attitude estimators.
  *
  */
-class VisionPoseEstimatePlugin : public plugin::PluginBase,
+class VisionPoseEstimatePlugin : public plugin::Plugin,
   private plugin::TF2ListenerMixin<VisionPoseEstimatePlugin>
 {
 public:
-  VisionPoseEstimatePlugin()
-  : PluginBase(),
-    sp_nh("~vision_pose"),
+  VisionPoseEstimatePlugin(plugin::UASPtr uas_)
+  : Plugin(uas_, "vision_pose"),
     tf_rate(10.0)
-  {}
-
-  void initialize(UAS & uas_) override
   {
-    PluginBase::initialize(uas_);
+    enable_node_watch_parameters();
 
     bool tf_listen;
 
     // tf params
-    sp_nh.param("tf/listen", tf_listen, false);
-    sp_nh.param<std::string>("tf/frame_id", tf_frame_id, "map");
-    sp_nh.param<std::string>("tf/child_frame_id", tf_child_frame_id, "vision_estimate");
-    sp_nh.param("tf/rate_limit", tf_rate, 10.0);
-
-    if (tf_listen) {
-      ROS_INFO_STREAM_NAMED(
-        "vision_pose", "Listen to vision transform " << tf_frame_id <<
-          " -> " << tf_child_frame_id);
-      tf2_start("VisionPoseTF", &VisionPoseEstimatePlugin::transform_cb);
-    } else {
-      vision_sub = sp_nh.subscribe("pose", 10, &VisionPoseEstimatePlugin::vision_cb, this);
-      vision_cov_sub = sp_nh.subscribe(
-        "pose_cov", 10, &VisionPoseEstimatePlugin::vision_cov_cb,
-        this);
-    }
+    node_declate_and_watch_parameter(
+      "tf/listen", false, [&](const rclcpp::Parameter & p) {
+        auto tf_listen = p.as_bool();
+        if (tf_listen) {
+            RCLCPP_INFO_STREAM(
+            get_logger(),
+            "Listen to vision transform" << tf_frame_id <<
+              " -> " << tf_child_frame_id);
+            tf2_start("VisionPoseTF", &VisionPoseEstimatePlugin::transform_cb);
+          } else{
+            vision_sub = node->create_subscription<geometry_msgs::msg::PoseStamped>(
+              "~/pose", 10, std::bind(
+                &VisionPoseEstimatePlugin::vision_cb, this, _1));
+            vision_cov_sub = node->create_subscription<geometry_msgs::msg::PoseWithCovarianceStamped>(
+              "~/pose_cov", 10, std::bind(
+                &VisionPoseEstimatePlugin::vision_cov_cb, this, _1));
+          }
+        });
```

Comment:
NOTE: place "switching" parameter after it's config parameters. Now you would try to start TF with unset frame ids.
Also it's good to make those parameters to be possible to change in operation (you'd have to reset all things to initial state, then init subs//pubs again).